### PR TITLE
Fix client not online deploy error

### PR DESCRIPTION
**Pull request by yhwen**
Fixed the issue if a client in the deploy_map not online, the job failed to deploy and run issue.

### CODE REVIEW

**Review by nvkevlu — APPROVED** [review]

File: nvflare/private/fed/server/job_runner.py [hunk 1]
```diff
@@ -77,18 +77,19 @@ def _deploy_job(self, job: Job, sites: dict, fl_ctx: FLContext) -> str:
                         fl_ctx, f"Application {app_name} deployed to the server for run:{run_number}", fire_event=False
                     )
                     if not success:
-                        raise RuntimeError("Failed to deploy the App to the server")
+                        raise RuntimeError(f"Failed to deploy the App: {app_name} to the server")
```
Comment (chesterxgchen) [inline nvflare/private/fed/server/job_runner.py:80]: is it possible to tell user why the deployment failed, networking, configuration, etc. etc. 
instead generic deployment failed ?  

Some instructive message, so user knows what failed, and he has next thing to do: fix network, start server, change configuration, fix code etc.

File: nvflare/private/fed/server/job_runner.py [hunk 2]
```diff
@@ -101,7 +102,8 @@ def _deploy_clients(self, app_data, app_name, run_number, client_sites, engine):
         message.set_header(RequestHeader.APP_NAME, app_name)
         replies = self._send_to_clients(admin_server, client_sites, engine, message)
         if not replies:
-            raise RuntimeError("Failed to deploy the App to the clients")
+            display_sites = ",".join(client_sites)
```
Comment (chesterxgchen) [inline nvflare/private/fed/server/job_runner.py:105]: ditto

**Review by chesterxgchen — APPROVED** [review]
I think we need to gradually improve the error message to give user more detailed failure message, and instructive. For now, I am ok with approve the improvement

File: nvflare/private/fed/server/job_runner.py (hunk 1 above)
Comment (yhwen) [reply nvflare/private/fed/server/job_runner.py:80]: How the app deployment works is loading the App data from the Job, and sending thee data to the client to unzip to the workspace. At this point, the Job has passed the validation, the job data structure and configuration meta.json already been validated. So the deployment failure only means the App failed to be sent to the client.

**Pull request merged**